src/hotspot/share/memory/universe.cpp

```diff
@@ -1302,49 +1302,8 @@ void Universe::verify(VerifyOption option, const char* prefix) {
   }
 }
 
-static void log_cpu_time() {
-  LogTarget(Info, cpu) cpuLog;
-  if (!cpuLog.is_enabled()) {
-    return;
-  }
-
-  const double process_cpu_time = os::elapsed_process_cpu_time();
-  if (process_cpu_time == 0 || process_cpu_time == -1) {
-    // 0 can happen e.g. for short running processes with
-    // low CPU utilization
-    return;
-  }
-
-  const double gc_threads_cpu_time = (double) CPUTimeUsage::GC::gc_threads() / NANOSECS_PER_SEC;
-  const double gc_vm_thread_cpu_time = (double) CPUTimeUsage::GC::vm_thread() / NANOSECS_PER_SEC;
-  const double gc_string_dedup_cpu_time = (double) CPUTimeUsage::GC::stringdedup() / NANOSECS_PER_SEC;
-  const double gc_cpu_time = (double) gc_threads_cpu_time + gc_vm_thread_cpu_time + gc_string_dedup_cpu_time;
-
-  const double elasped_time = os::elapsedTime();
-  const bool has_error = CPUTimeUsage::Error::has_error();
-
-  if (gc_cpu_time < process_cpu_time) {
-    cpuLog.print("=== CPU time Statistics =============================================================");
-    if (has_error) {
-      cpuLog.print("WARNING: CPU time sampling reported errors, numbers may be unreliable");
-    }
-    cpuLog.print("                                                                            CPUs");
-    cpuLog.print("                                                               s       %%  utilized");
-    cpuLog.print("   Process");
-    cpuLog.print("     Total                        %30.4f  %6.2f  %8.1f", process_cpu_time, 100.0, process_cpu_time / elasped_time);
-    cpuLog.print("     Garbage Collection           %30.4f  %6.2f  %8.1f", gc_cpu_time, percent_of(gc_cpu_time, process_cpu_time), gc_cpu_time / elasped_time);
-    cpuLog.print("       GC Threads                 %30.4f  %6.2f  %8.1f", gc_threads_cpu_time, percent_of(gc_threads_cpu_time, process_cpu_time), gc_threads_cpu_time / elasped_time);
-    cpuLog.print("       VM Thread                  %30.4f  %6.2f  %8.1f", gc_vm_thread_cpu_time, percent_of(gc_vm_thread_cpu_time, process_cpu_time), gc_vm_thread_cpu_time / elasped_time);
-
-    if (UseStringDeduplication) {
-      cpuLog.print("       String Deduplication       %30.4f  %6.2f  %8.1f", gc_string_dedup_cpu_time, percent_of(gc_string_dedup_cpu_time, process_cpu_time), gc_string_dedup_cpu_time / elasped_time);
-    }
-    cpuLog.print("=====================================================================================");
-  }
-}
 
 void Universe::before_exit() {
-  log_cpu_time();
```

Comment:
Why did you move `log_cpu_time()`? During the review of CPUTimeUsage refactor (https://github.com/openjdk/jdk/pull/26621) we discussed this choice. Given that it still includes more than just GC I think it should stay in `Universe`. Also the PR title does not reflect that it would include a refactor of CPUTimeUsage.